viamonvif/device/device_test.go

```diff
@@ -91,3 +92,106 @@ func readBody(r *http.Request) (string, error) {
 	}
 	return buf.String(), nil
 }
+
+func TestTLSVerificationConfig(t *testing.T) {
+	logger := logging.NewTestLogger(t)
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		body, err := readBody(r)
+		if err != nil {
+			t.Fatalf("failed to read body: %v", err)
+		}
+
+		if strings.Contains(r.Header.Get("Content-Type"), "soap") &&
+			strings.Contains(body, "GetCapabilities") {
+			w.Write([]byte(`<?xml version="1.0" encoding="UTF-8"?>
+				<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://www.w3.org/2003/05/soap-envelope">
+					<SOAP-ENV:Body>
+						<GetCapabilitiesResponse>
+							<Capabilities>
+								<Media>
+									<XAddr>http://example.com/onvif/media</XAddr>
+								</Media>
+							</Capabilities>
+						</GetCapabilitiesResponse>
+					</SOAP-ENV:Body>
+				</SOAP-ENV:Envelope>`))
+		}
+	}))
+	defer server.Close()
+
+	testCases := []struct {
+		name                     string
+		isLocal                  bool
+		skipLocalTLSVerification bool
+		expectSkipVerify         bool
+	}{
+		{
+			name:                     "IP local, skip enabled",
+			isLocal:                  true,
+			skipLocalTLSVerification: true,
+			expectSkipVerify:         true,
+		},
+		{
+			name:                     "IP local, skip disabled",
+			isLocal:                  true,
+			skipLocalTLSVerification: false,
+			expectSkipVerify:         false,
+		},
+		{
+			name:                     "IP public, skip enabled",
+			isLocal:                  false,
+			skipLocalTLSVerification: true,
+			expectSkipVerify:         false,
+		},
```

Comment:
Do we want this case at all?

In my mind - the case that is bad user experience is a local IP address which we are being overly secure with on a person's private network. But this test case seems slightly too permissive to me.

Reply:
Can you walk me through the expected behaviour of the public IPs only - I want to make sure we're testing the correct things.

Reply:
For "IP public, skip enabled," this test should pass without any errors, confirming that the device can connect to the public IP **without** skipping TLS verification. Basically making sure that the http client isn't bypassing the verification for it, therefore asserting `expectSkipVerify:         false`

For "IP public, skip disabled," this is the most trivial case where we assert that we aren't skipping TLS verification even when the flag is disabled.